domain.go

```diff
@@ -36,15 +38,91 @@ type DomainUpdateInput struct {
 }
 
 func (s *DomainId) ChildSystems(client *Client, variables *PayloadVariables) (*SystemConnection, error) {
-	return &SystemConnection{}, nil
+	var q struct {
+		Account struct {
+			Domain struct {
+				ChildSystems SystemConnection `graphql:"childSystems(after: $after, first: $first)"`
+			} `graphql:"domain(input: $domain)"`
+		}
+	}
+	if s.Id == "" {
+		return nil, fmt.Errorf("Unable to get Systems, invalid domain id: '%s'", s.Id)
+	}
+	if variables == nil {
+		variables = client.InitialPageVariablesPointer()
+	}
+	(*variables)["domain"] = IdentifierInput{
+		Id: s.Id,
+	}
+
+	if err := client.Query(&q, *variables, WithName("DomainChildSystemsList")); err != nil {
+		return nil, err
+	}
+	for q.Account.Domain.ChildSystems.PageInfo.HasNextPage {
+		(*variables)["after"] = q.Account.Domain.ChildSystems.PageInfo.End
+		resp, err := s.ChildSystems(client, variables)
+		if err != nil {
+			return nil, err
+		}
+		q.Account.Domain.ChildSystems.Nodes = append(q.Account.Domain.ChildSystems.Nodes, resp.Nodes...)
+		q.Account.Domain.ChildSystems.PageInfo = resp.PageInfo
+		q.Account.Domain.ChildSystems.TotalCount += resp.TotalCount
+	}
+	return &q.Account.Domain.ChildSystems, nil
 }
 
 func (s *DomainId) Tags(client *Client, variables *PayloadVariables) (*TagConnection, error) {
-	return &TagConnection{}, nil
+	var q struct {
+		Account struct {
+			Domain struct {
+				Tags TagConnection `graphql:"tags(after: $after, first: $first)"`
+			} `graphql:"domain(input: $domain)"`
+		}
+	}
+	if s.Id == "" {
+		return nil, fmt.Errorf("Unable to get Tags, invalid domain id: '%s'", s.Id)
+	}
+	if variables == nil {
+		variables = client.InitialPageVariablesPointer()
+	}
+	(*variables)["domain"] = IdentifierInput{
+		Id: s.Id,
+	}
+
+	if err := client.Query(&q, *variables, WithName("DomainTagsList")); err != nil {
+		return nil, err
+	}
+	for q.Account.Domain.Tags.PageInfo.HasNextPage {
+		(*variables)["after"] = q.Account.Domain.Tags.PageInfo.End
+		resp, err := s.Tags(client, variables)
+		if err != nil {
+			return nil, err
+		}
+		q.Account.Domain.Tags.Nodes = append(q.Account.Domain.Tags.Nodes, resp.Nodes...)
+		q.Account.Domain.Tags.PageInfo = resp.PageInfo
+		q.Account.Domain.Tags.TotalCount += resp.TotalCount
+	}
+	return &q.Account.Domain.Tags, nil
 }
 
 func (s *DomainId) AssignSystem(client *Client, systems ...string) error {
-	return nil
+	var m struct {
+		Payload struct {
+			Domain Domain
+			Errors []OpsLevelErrors
+		} `graphql:"domainChildAssign(domain:$domain, childSystems:$childSystems)"`
+	}
+	childSystems := []IdentifierInput{}
+	for _, system := range systems {
+		systemIdentifier := NewIdentifier(system)
```

Comment:
was mainly concerned here because systems is a list of identifiers.  Tried updating with this.

Reply:
Might be handy to wrap this up into a utility function that takes in a `[]string` and return `[]IdentifierInput`. I suspect we'll need to do this kind of thing in the future with Infra and Custom Objects

Reply:
Hmm, just realized you were saying to do this for a list but I did it for an individual string.  Adding this helper function now.  I can delete the one that we don't need.

Reply:
Ok, yeah don't need a helper function for single value but created one for the list case and implemented here.